linux_os/guide/system/accounts/accounts-pam/set_password_hashing_algorithm/set_password_hashing_algorithm_libuserconf/oval/shared.xml

```diff
@@ -1,21 +1,29 @@
 <def-group>
-  <definition class="compliance" id="set_password_hashing_algorithm_libuserconf" version="1">
+  <definition class="compliance" id="{{{ rule_id }}}" version="1">
     {{{ oval_metadata("The password hashing algorithm should be set correctly in /etc/libuser.conf.") }}}
     <criteria operator="AND">
-      <criterion test_ref="test_etc_libuser_conf_cryptstyle" />
+      <criterion test_ref="test_set_password_hashing_algorithm_libuserconf" />
     </criteria>
   </definition>
 
-  <ind:textfilecontent54_test check="all" check_existence="at_least_one_exists"
-  comment="The password hashing algorithm should be set correctly in /etc/libuser.conf"
-  id="test_etc_libuser_conf_cryptstyle" version="1">
-    <ind:object object_ref="object_etc_libuser_conf_cryptstyle" />
+  <ind:textfilecontent54_test id="test_set_password_hashing_algorithm_libuserconf" version="2"
+    check="all" check_existence="at_least_one_exists"
+    comment="check if /etc/libuser.conf hashing algorithm option is correct">
+    <ind:object object_ref="object_set_password_hashing_algorithm_libuserconf"/>
+    <ind:state state_ref="state_set_password_hashing_algorithm_libuserconf"/>
   </ind:textfilecontent54_test>
 
-  <ind:textfilecontent54_object comment="The password hashing algorithm should be set correctly in /etc/libuser.conf"
-  id="object_etc_libuser_conf_cryptstyle" version="1">
+  <ind:textfilecontent54_object id="object_set_password_hashing_algorithm_libuserconf" version="2"
+    comment="only one hashing algorithm option for pam_unix.so is found in /etc/pam.d/password-auth">
     <ind:filepath>/etc/libuser.conf</ind:filepath>
-    <ind:pattern operation="pattern match">^[\s]*crypt_style[\s]+=[\s]+(?i)sha512[\s]*$</ind:pattern>
+    <ind:pattern operation="pattern match">^[\s]*crypt_style[\s]*=[\s]*(\w*)[\s]*$</ind:pattern>
```

Comment:
Is the space around the equal sign mandatory or optional?

Reply:
Optional. I don't know why the check was requiring this. It is a good practice, however I didn't find anything invalidating the absence of spaces.